### ISSUE TITLE
[SUGGESTION] Core gameplay issues 

### ISSUE THREAD

**Issue by C0rn3j** [Shinoow/AbyssalCraft#179]
After playing through the mod, I've found the following issues with the core gameplay -

1) The world gen in the overworld is way too invasive. There's a ton of AC biomes over my world and it's just not pretty. And even then I have massive trouble finding the biome I need to get the items I require.

I don't want to have a ton of biomes which are unusable because of all the blidness and general ugliness(imo) of the grass(but mostly the blindness).

2)To get the items needed from the ocean biome - it is just WAY TOO GRINDY. Even with veinminer set to take in all blocks it's too much.

What I suggest is remove the AC biomes and ores from the overworld(perhaps keep the shoggoths there?), add a new dimension you need to access first to get the starting items(coralium, shadow gems) and make it way less grindy. I wouldn't even mind looking for the hills(or mountains?) biome you need for the big mobs to spawn for a few minutes, but I really mind looking for them for hours.

Maybe make a compass that points to the biome too?

Anyway, I enjoyed the mod, and I didn't mind grinding for the energy needed to progress further, even though it could be made a bit easier. 

Tl;dr Move biome generation from overworld to a new dimension and add a compass to find Mountains biome, move Coralium to the new dimension too in bigger numbers.

**Comment by Viatos** [comment]
You can decrease the biome frequency in the config, and shut off biomes you don't want entirely. I'd suggest leaving basic Darklands, Darklands Mountains, and Coralium-Infested Swamps on, though, they all have stuff you'll want.

What else do you need from the ocean aside from Coralium?

**Comment by C0rn3j** [comment]
>What else do you need from the ocean aside from Coralium?

Nothing, just the coralium ^^

**Comment by Shinoow** [comment]
> The world gen in the overworld is way too invasive. There's a ton of AC biomes over my world and it's just not pretty. And even then I have massive trouble finding the biome I need to get the items I require.

As Viatos said above, you can configure the chance of the biomes generating, while also having the option to disable half of the biomes in favor of the ones that really necessary among them.

> I don't want to have a ton of biomes which are unusable because of all the blidness and general ugliness(imo) of the grass(but mostly the blindness).

There's a config option to disable the blindness, so you always have the option to toggle that. Alternatively, wearing any helmet from the mod stops the blindness, so that's also an option. If you don't like the grass, you can make a resourcepack that replaces the texture (or suggest a change of texture/color/whatever to it).

> 2)To get the items needed from the ocean biome - it is just WAY TOO GRINDY. Even with veinminer set to take in all blocks it's too much.

Well, you've never actually been forced to mine underneath an ocean for Coralium. The ore generates in swamps as well (not to mention that there's a lot more of it in the Coralium Infested Swamp, hence the name). Going after it in ocean biomes has always been a optional thing, as is going through swamps. Heck, you could even rely entirely on dungeon chests (since it can be found as loot in them, among Shadow Fragments/Shards/Gems, which themselves can be found in nearly all of the loot tables for those type of structures).

> What I suggest is remove the AC biomes and ores from the overworld(perhaps keep the shoggoths there?), add a new dimension you need to access first to get the starting items(coralium, shadow gems) and make it way less grindy. I wouldn't even mind looking for the hills(or mountains?) biome you need for the big mobs to spawn for a few minutes, but I really mind looking for them for hours.

I had someone request that the overworld biomes were turned into a dimension (not sure where they asked it, nor when, since after searching various places for an hour, I've found nothing), but the short answer is no. To some degree, moving them into a separate dimension would be more grindy (since the sole purpose of said dimension would be grinding for resources). Then there's the extra computer resources of having yet another dimension (and this one would potentially be in the category of dimensions you always have loaded, since it's pretty much an extension to the overworld), and the quantity of new content and adjustments to the mod to in order for this to actually fit in.

> Maybe make a compass that points to the biome too?

Isn't there already several mods adding just that? The general rarity of a biome varies per seed (biome weight is also a factor here, and it's fully configurable), which means the biomes can be rare in some cases, while a lot more common in other (common is a loose term here, since if the biome weight is very low, spawning within close proximity of one is luck in that case). If I can get some more opinions on biome rarity, this could potentially become a thing in the mod.

---

TL;DR biome weights are configurable (eg. you can control the chance of each biome generating), and half of them are mostly aesthetic, meaning you can disable them without any issues. Blidness can be mitigated, and the Darklands Grass can be changed if some suitable change is suggested. Coralium generates in more places than oceans, check swamps. Not gonna move the biomes into their own dimension, read the paragraph for reasons why. With more opinions, a biome compass could become a thing, otherwise there should already be a few mods with a function like that.

**Comment by C0rn3j** [comment]
>Then there's the extra computer resources of having yet another dimension (and this one would potentially be in the category of dimensions you always have loaded, since it's pretty much an extension to the overworld)

Well, if there's no players in it it wouldn't be loaded ^^

> Blindness

The Twilight Forest mod has a few debuffing biomes, where if you do a certain thing you will no longer get debuffed in the biome. How about adding something you can do that makes you being immune to the biome blindness? It'd be much more preferable than wearing a mod-specific helmet.

> Coralium

I've never realized that the infested swamps have much more Coralium. Adding the compass and informing the player why they'd want to go to the certain biome on selection would make it much easier to figure it out.

> Biome frequency

Mostly your suggestions are "change the default config/install another mod". I'm using a server with over 150 mods, and there is only one config I've changed for one mod. 

I'd say that having a default config that majority of the playerbase enjoys is a good thing. (I'm assuming here, but everyone I played with complained about the frequency of the biomes).

**Comment by Viatos** [comment]
Not to whiteknight here, but if you're running 150 mods and didn't bother to configure them, IMHO you can't really expect a tailored experience. The thing that you want - for biomes to spawn differently - is literally changing 8 small numbers in a file you can open with Notepad. It's labeled for you and everything.

Also you can turn off Blindness - again, in the config file. The purpose of those files is to let you "have it your way" which is why so many mods use them. Arguing that the config doesn't DEFAULT to your preference is like arguing that cheeseburgers should default to having onions. They'll put the onions on, dude.

I DO agree that the biomes default to showing up more often than I want - I kinda hate the way Darklands look, garish like Taint was garish in Thaumcraft - but on the other hand you need the stuff in them to get started with the mod, and if you disable some of them the effective frequency goes way down. I just keep Mountains, Swamps, and Darklands at low Frequency and in a map I've got explored for just about two thousand units square, I've got two Swamps and two Darklands.

**Comment by C0rn3j** [comment]
> if you disable some of them the effective frequency goes way down

Which is not an issue if you have a way to find them - e.g. the compass.

> if you're running 150 mods and didn't bother to configure them, IMHO you can't really expect a tailored experience

I'm happy with how the mods are(granted I don't fully know all of them, and some not at all), I don't demand a super tailored experience without configuring the heck out of something, I just want something acceptable, and that is not achievable with the default config - not for me or any of my friends who play with me.

**Comment by Viatos** [comment]
Adjust the config.

**Comment by Shinoow** [comment]
The thing about the config defaults for the biome weights is that the current default value they have (5, started out as 10) was decided after a conversation with a lot of people pitching in (I have no idea where or when, but it took place somewhere on the internet, and should be possible to trace down). If more people reached out to me about the biome frequency still being way too high, perhaps the default could be adjusted once more (but since the only complaints I generally get regarding the biome is the sizes, which I have absolutely no control over what so ever, no changes has been done since the move from 10 to 5).
If I can get some opinion on a new value that fits them more, then I can set the default to that value instead of the current.

**Comment by LatentDesire** [comment]
OP not bothering to adjust config (that would deal with their specific issue) because they want things to be their way by default is probably a big contributor to mod authors leaving. 

I remember watching the passive flower decay debacle evolve over time with Vazkii's mod Botania. Vazkii practically bent over backwards to accommodate people who refused to progress further than the passive flowers (And then complained about the mod being to slow or grindy). When decay was introduced to encourage using better generation methods, people flipped their shit that they couldn't just spam like a chunk worth of dayblooms and never have to deal with the other generation methods. Eventually they added a config to turn off decay (despite being against it.) and it still wasn't enough. (Then there were special mechanics they introduced to bypass the decay in their own mod outside the config)  It got so bad that in the latest version (I think) Dayblooms and nightshade are no longer even exist.

I commend Vazkii for finally putting their vision of the mod as priority, and no longer hostage to how the "fans" think that their mod should work.

**Comment by C0rn3j** [comment]
@LatentDesire
I take it you didn't read the whole thread.

@Shinoow
The thing is, if you make the biomes more pleasant to deal with, changing the spawn rate of them might not even be an issue.

Example - 
Create an achievement that's not too hard to obtain that will make you invulnerable to blindness after you obtain it. Make it clear that this is why you keep getting blinded in AC biomes.(message when user gets blinded).

Make it easier to understand/document better that coralium in the swamp biomes is way easier to get than the oceans.

Make it known that the user likely wants a compass mod - [make this mod](https://minecraft.curseforge.com/projects/natures-compass)(for example) an optional library on CurseForge and mention it in the main page.

Personally, I don't love the biomes. Perhaps make it an end-game feature to choose whether you want to convert the purple biomes to regular/some other biomes, and possibly the shoggoth lairs too? The user will already have access to the resource in the dimensions at that point ^^

**Comment by Viatos** [comment]
I think acknowledging how incredibly dumb it is that the OP refuses to change their configs shows great reading comprehension, personally.

The blindness can be turned off. I definitely DO turn it off because it's really annoying, and I wouldn't be sad to see it go, to be fair. 

My personal complaint is that I think the Darklands biomes are ugly/garish and the tree effect doesn't look good in aggregate - honestly I don't think those kinds of effects ever look good in something you theoretically want to build with - but I also acknowledge that that's an aesthetic preference and maybe some people enjoy that godawful shade of purple. I'd like to see a shift to, say, a dark blue-green that would work alongside instead of against Minecraft's normative palette, and for the trees to be more black-bark tangled-canopy trees that could form actual forests instead of the weird GIF scarecrows they are now. I'd also like to see the mountains have more visual variation, ice and snow, other terrain besides just that black stone. But I can also just not settle near them, and I think the swamps are pretty cool if too oppressive to ever, say, build anything in.

Better documentation is always worth pursuing but FWIW I definitely took note that coralium is easier to find in the swamp literally called Coralium-Infested without needing to ask people. There is a getting started section in the Necronomicon that mentions it, I think, it's been a little while since I've paged through it.

**Comment by LatentDesire** [comment]
While I can see how the blindness effect can be annoying, it's plainly obvious to me why such a thing exists. Aside from the color, and name, the blindness effect drives home that these are not nice places to be, let alone live. I don't remember offhand if the Necronomicon has a lore behind what it is that makes the darklands dark. Weather it's the land that attracts the creatures, or if the creatures effect the land. 

Using thaumcraft's taint as an example, we know why tainted land biomes exist in the lore. We also can effect changes on said biomes, however slowly. While being able to effect changes on something which previously we had no control may be against some sense of cosmic horror, the fact of some of the bosses can be defeated (and you can amass such power to do so) also goes against this. So I do agree that at some point in the progression being able to lighten/uninfest the biomes might be a good feature. 

Another possible option is for the effect of the darklands not being the vanilla blindness effect. For example, perhaps a consistent dark fog that effects you once you enter. Possibly tied with an effect that looks as if a liquid darkness is bubbling up from the ground itself to surround you from time to time. (Now that I type that, it sounds incredibly awesome.) The liquid darkness effect can be what the helmets protect from. 

Also, since I'm sure some people would like to use other armors and such, perhaps bringing in bauble integration would be a good idea. Some bauble(s) aside from a cool unique effect can possibly protect from the liquid darkness/other effects.

**Comment by Shinoow** [comment]
![2017-01-14_22 22 04](https://cloud.githubusercontent.com/assets/6419600/21958198/bdc639a4-daa8-11e6-9dcb-59a89ff55669.png)
![2017-01-14_22 25 33](https://cloud.githubusercontent.com/assets/6419600/21958201/d098f1d4-daa8-11e6-97f0-2bd2fc2bda3e.png)
![2017-01-14_22 42 00](https://cloud.githubusercontent.com/assets/6419600/21958260/b6b76ece-daaa-11e6-9160-ae468179ac0f.png)
![2017-01-14_22 46 16](https://cloud.githubusercontent.com/assets/6419600/21958287/4e2a86b0-daab-11e6-8e6e-3c8eb11320a3.png)
Started poking around with changes to the color palette.

**Comment by Shinoow** [comment]
@C0rn3j 

>The thing is, if you make the biomes more pleasant to deal with, changing the spawn rate of them might not even be an issue.
Example -
Create an achievement that's not too hard to obtain that will make you invulnerable to blindness after you obtain it. Make it clear that this is why you keep getting blinded in AC biomes.(message when user gets blinded).

The message part would need the whole blindness part to be changed (since the frequency when the potion effect is applied is randomized, leading to the chat being spammed with these messages). What @LatentDesire suggested might be a better thing (or just remove the whole thing).

>Make it easier to understand/document better that coralium in the swamp biomes is way easier to get than the oceans.

Quoting @Viatos here, the biome is called _Coralium Infested Swamp_, wouldn't that alone be a implication that there might be larger quantities of Coralium there? Since the information in the Necronomicon will be rewritten altogether in the near future, a lot of details will be emphasized.

>Make it known that the user likely wants a compass mod - make this mod(for example) an optional library on CurseForge and mention it in the main page.

So basically, you're telling me to drop the biome weights to a point where the biomes hardly ever spawn, then tell the user to install another mod just to find them? Sounds like I'd be better off just leaving the weights where they're currently at.

>Personally, I don't love the biomes. Perhaps make it an end-game feature to choose whether you want to convert the purple biomes to regular/some other biomes, and possibly the shoggoth lairs too? The user will already have access to the resource in the dimensions at that point ^^

As some of the biomes have a vanilla equivalent, having the terrain revert to that (or transform to some degree) would be doable. Since the lairs are just randomly generated structures, there wouldn't really be any efficient way to locate them.

@Viatos 
>My personal complaint is that I think the Darklands biomes are ugly/garish and the tree effect doesn't look good in aggregate - honestly I don't think those kinds of effects ever look good in something you theoretically want to build with - but I also acknowledge that that's an aesthetic preference and maybe some people enjoy that godawful shade of purple. I'd like to see a shift to, say, a dark blue-green that would work alongside instead of against Minecraft's normative palette, and for the trees to be more black-bark tangled-canopy trees that could form actual forests instead of the weird GIF scarecrows they are now. I'd also like to see the mountains have more visual variation, ice and snow, other terrain besides just that black stone. But I can also just not settle near them, and I think the swamps are pretty cool if too oppressive to ever, say, build anything in.

Finally someone who actually suggests a change in color instead of just complaining about the biomes sole existence, or either suggesting removing them altogether or move them to a place where nobody can ever see them!
Reducing the amount of randomization on the trees would bring them closer to something canopy-looking (provided there's an adequate amount of branches), but the current design was put there to give them a less "normal tree" look (since they originally looked like vanilla trees).
I could add a biome mutation for the mountain that makes it snow there (with surface changes), alternatively apply height-dependent changes to it, so that the terrain gradually changes the higher it generates.
Due to the swamps being a source of physics-violating things, that alone makes it more of a challenge to settle down in that area.

@LatentDesire 
>While I can see how the blindness effect can be annoying, it's plainly obvious to me why such a thing exists. Aside from the color, and name, the blindness effect drives home that these are not nice places to be, let alone live. I don't remember offhand if the Necronomicon has a lore behind what it is that makes the darklands dark. Weather it's the land that attracts the creatures, or if the creatures effect the land.

I want to recall writing down lore to it at some point, but I'm not sure if said lore can be found anywhere. With the various shadows being present, the Dark Realm would have some involvement in those biomes being the way they are. I plan on writing down some actual lore for it at some point.

>Using thaumcraft's taint as an example, we know why tainted land biomes exist in the lore. We also can effect changes on said biomes, however slowly. While being able to effect changes on something which previously we had no control may be against some sense of cosmic horror, the fact of some of the bosses can be defeated (and you can amass such power to do so) also goes against this. So I do agree that at some point in the progression being able to lighten/uninfest the biomes might be a good feature.

In a lot of cases here, you're not really killing the bosses. There's a unfinished chapter meant to be in the Abyssalnomicon explaining this, but the gist if Asorah is never alive (dead long before the player ever reaches the Abyssal Wasteland, hence the ritual, which reanimates a manifestation of the dragon), Cha'garoth is immortal (every Dread Spawn or Spawn of Cha'garoth is a small piece of him, and once the current Dreadbeast dies, a Lesser Dreadbeast will grow even larger, taking the place as the new Dreadbeast, thus allowing Cha'garoth to continue existing), J'zahar is not _really_ J'zahar (this part is already explained in the "villain speech" held during the death animation), Sacthoth is the step after a Shadow Beast (and all living shadows are manifestations from the souls of beings that have existed at some point in time, among those beings Asorah, and the shadows themselves depart back to the Dark Realm whenever their "host" or their current form "dies" in any plane, whereas Sacthoth is materialized from the energy unleashed in the explosion that summons him). But to purge the biomes of the darkness would fit quite well.

>Another possible option is for the effect of the darklands not being the vanilla blindness effect. For example, perhaps a consistent dark fog that effects you once you enter. Possibly tied with an effect that looks as if a liquid darkness is bubbling up from the ground itself to surround you from time to time. (Now that I type that, it sounds incredibly awesome.) The liquid darkness effect can be what the helmets protect from.

That would be a perfect replacement for the blindness (other alternative being to remove the effect altogether), and could also be a pretty good aesthetic effect alongside the insanity mechanism (when that makes it's appearance). Tying this to an achievement triggered at some point could also be a suitable permanent solution instead of forcing the headgear.

>Also, since I'm sure some people would like to use other armors and such, perhaps bringing in bauble integration would be a good idea. Some bauble(s) aside from a cool unique effect can possibly protect from the liquid darkness/other effects.

Baubles in general might be something I'll implement at some point, or at least some passive item that has some effect when either held or in the inventory (maybe hotbar). This gives yet another option to dodge the darkness.

**Comment by LatentDesire** [comment]
@Shinoow Ah, my mistake on the lore. 

I do rather like my bubbling liquid darkness idea, but it's got me thinking, would liquid darkness be an actual liquid (in addition to an effect) in the darklands and by extension the Dark realm? I feel like being submerged in it will cause immediate blindness. What other uses it could have escapes me. Though seeing as how darkness is immaterial by it's nature, it would be interesting to see how antimatter would interact with it. Solid darkness? Something stranger? I'm sure a mad mind could come up with all sorts of possibilities and uses.

**Comment by Shinoow** [comment]
It could have the ability to convert entities dropping into it into shadow mobs. Although, that might be a bit too simple, but it could follow the scheme of Liquid Coralium and Liquid Antimatter in that situation (eg. apply a potion effect that has a chance of converting the killed target into something), so that you have a chance of spawning the various shadows by making things die to this liquid.
In terms of physical properties, it would probably be of a gaseous sort, where it doesn't affect your mobility, but still does it's thing to you if you come in contact with it.

**Comment by LatentDesire** [comment]
"Oh, it looks like there's some water down there. I'll just jump in and…" *Splat!*

**Comment by Shinoow** [comment]
With 1.9.3.10 being out, I think most of the stuff here is covered. If there's still something left, leave the issue open, otherwise close it.

**Comment by C0rn3j** [comment]
I didn't see any biome frequency related changes so I'd add the compass mod as an optional library.

Except for that, I think my issue is pretty solved, thanks everyone.
